analytics/src/analytics/integrations/etldb/main.py

```diff
@@ -17,35 +19,38 @@
 VERBOSE = False
 
 
-def init_db() -> None:
+def initialize_database() -> None:
     """Initialize etl database."""
-    # define the path to the sql file
-    parent_path = Path(__file__).resolve().parent
-    sql_path = f"{parent_path}/create_etl_db.sql"
-
-    # read sql file
-    with open(sql_path) as f:
-        sql = f.read()
-
-    # execute sql
-    try:
-        db = EtlDb()
-        cursor = db.connection()
-        cursor.execute(
-            text(sql),
-        )
-        db.commit(cursor)
-    except (
-        RuntimeError,
-        ProgrammingError,
-        OperationalError,
-        InsufficientPrivilege,
-    ) as e:
-        message = f"FATAL: Failed to initialize db: {e}"
-        raise RuntimeError(message) from e
+    # get connection to database
+    etldb = EtlDb()
+    current_version = etldb.get_schema_version()
+    print(f"current schema version: {current_version}")
+
+    # get all sql file paths and respective version numbers
+    sql_file_path_map = get_sql_file_paths()
+    all_versions = sorted(sql_file_path_map.keys())
+
+    # iterate sql files
+    for next_version in all_versions:
+        if next_version <= current_version:
+            continue
+        # read sql file
+        with open(sql_file_path_map[next_version]) as f:
+            sql = f.read()
+            # execute sql
+            print(f"applying updates for schema version: {next_version}")
+            cursor = etldb.connection()
+            cursor.execute(
+                text(sql),
+            )
+            # commit changes
+            etldb.commit(cursor)
+            # bump schema version number
+            etldb.set_schema_version(next_version)
+            current_version = next_version
```

Comment:
Minor nit: I think you can get rid of this? If there are multiple files to process, current_version will always be the last one processed (before the current run), so you just want everything after than anyways.

Reply:
This protects against sql files that duplicate the 4 digit prefix, e.g. 
`0007_do_stuff.sql`
`0007_do_more_stuff.sql`

Reply:
I don't have a strong preference about this but want to check my understanding on how the code is supposed to behave in the case when there are multiple migration scripts with the same version:

- Are just skipping over duplicate versions? Or raising an error? I imagine we want to due the latter in get_sql_file_paths()
- How are we ordering between versions if there are duplicates (if we're not raising an error)?
- Also if we're fetching the path from sql_file_path_map then I don't actually think we'd actually have duplicate versions since it's keyed on version.

```python
for filename in filename_list:
    # validate filename format
    if pattern.match(filename):
        version = int(filename[:4])
        result[version] = f"{sql_file_directory}/{filename}"
```

Reply:
Yeah, I would agree that I would expect a fatal error if the files don't match expectations, not running one? of the offending files per alpha sorting and silently ignoring the rest? if I'm reading this right?

Reply:
Good catch. I'll raise an exception if there are duplicate version, rather than silently ignore them.

Reply:
Done - please see latest commits